tests/Spec.hs

```diff
@@ -1877,10 +1894,55 @@ main = hspec $ do
                         False
         , ALeaveComment (PullRequestId 3) "<!-- Hoff: ignore -->\nSpeculatively rebased as 3b2 behind 2 other PRs, waiting for CI …"
         , ALeaveComment (PullRequestId 3) "<!-- Hoff: ignore -->\n[CI job :yellow_circle:](example.com/3b2) started."
-        , ATryPromote (Branch "fst") (Sha "1b2")
+        , ATryForcePush (Branch "fst") (Sha "1b2")
+        , ATryPromote (Sha "1b2")
         , ACleanupTestBranch (PullRequestId 1)
         ]
 
+    it "does not promote if timeout has not expired" $
+      let prId = PullRequestId 1
+          state = singlePullRequestState prId (Branch "p") masterBranch (Sha "abc1234") "tyrell"
+          results = defaultResults { resultIntegrate = [Right (Sha "def2345")] }
+
+          event =
+            [ CommentAdded prId "deckard" "@bot merge"
+            , BuildStatusChanged (Sha "def2345") "default" (Project.BuildStarted "example.com/1b2")
+            , BuildStatusChanged (Sha "def2345") "default" Project.BuildSucceeded
+            , ClockTick (Time.addTime testTime 100)
+            ]
+          (_, actions) = runActionCustom results $ handleEventsTest event state
+       in actions `shouldBe`
+            [ AIsReviewer (Username "deckard")
+            , ALeaveComment prId "<!-- Hoff: ignore -->\nPull request approved for merge by @deckard, rebasing now."
+            , ATryIntegrate "Merge #1: Untitled\n\nApproved-by: deckard\nAuto-deploy: false\n" (PullRequestId 1, Branch "refs/pull/1/head", Sha "abc1234") [] False
+            , ALeaveComment prId "<!-- Hoff: ignore -->\nRebased as def2345, waiting for CI \8230"
+            , ALeaveComment prId "<!-- Hoff: ignore -->\n[CI job :yellow_circle:](example.com/1b2) started."
+            , ATryForcePush (Branch "p") (Sha "def2345")
+            ]
+
+    it "promotes if timeout has expired" $
```

Comment:
Interestingly, by detaching the rebase (force-)push from the push to the 'main'
branch into separate steps. It is now possible that hoff handles, and perhaps
rebases any other PR that may have been `@bot merge`d. If i'm not mistaken the
current implementation will simply cause one of the two merge's to fail. How
this interacts with the merge train functionality is also quite confusing.

Can you add a testcase for this, and see if this is a problem?

Reply:
Was explained offline. What happens is that the subsequent `@bot merge` wil put the PR in the merge train and will wait until the previous merge gets pushed to master.